lib/modules/manager/gitlabci-include/extract.ts

```diff
@@ -73,22 +16,14 @@ export function extractPackageFile(
     platform === 'gitlab' && endpoint
       ? [endpoint.replace(regEx(/\/api\/v4\/?/), '')]
       : null;
+
   try {
-    // TODO: use schema (#9610)
-    const docs = parseYaml<GitlabPipeline>(content, {
-      uniqueKeys: false,
-    });
-    for (const doc of docs) {
-      if (is.object(doc)) {
-        const includes = getAllIncludeProjects(doc);
-        for (const includeObj of includes) {
-          const dep = extractDepFromIncludeFile(includeObj);
-          if (registryUrls) {
-            dep.registryUrls = registryUrls;
-          }
-          deps.push(dep);
-        }
+    const docs = parseYaml(content, { uniqueKeys: false });
+    for (const dep of GitlabDocumentArray.parse(docs)) {
```

Comment:
pass schema via `customSchema` option to `parseYaml`

Reply:
No, its type inference doesn't work correctly. In order to fix it, I need first write schemas for all cases that pass generic unchecked type `T` to the `parseSchema`. Until then, `customSchema` is discouraged.

TLDR; if I do, the type of `docs` will be `unknown[]`